README.md

```diff
@@ -1,102 +1,269 @@
-instability. Features hoped-for soon include:
+fn ui_builder() -> impl Widget<u32> {
+    // The label text will be computed based dynamically based on the current locale and count
+    let text =
+        LocalizedString::new("hello-counter").with_arg("count", |data: &u32, _env| (*data).into());
+    let label = Label::new(text);
+    let button = Button::new("increment", |_ctx, data, _env| *data += 1);
 
-  - [ ] More of the basic widgets
-  - [ ] Incremental presentation
-  - [ ] Some simple theming support
+    let mut col = Column::new();
+    col.add_child(Align::centered(Padding::new(5.0, label)), 1.0);
+    col.add_child(Padding::new(5.0, button), 1.0);
+    col
+}
+```
 
-The biggest single obstacle to porting is 2d graphics, as druid currently
-uses Direct2D (and DirectWrite for text). One way forward is to create a
-[2d graphics] abstraction.
+Check out the [the examples folder] for a more comprehensive demonstration of
+druid's existing functionality and widgets.
 
 ## Screenshots
 
 #### Linux
-![](https://raw.githubusercontent.com/xi-editor/druid/screenshots/images/calc.png)
-![](https://raw.githubusercontent.com/xi-editor/druid/screenshots/images/custom_widget.png)
-![](https://raw.githubusercontent.com/xi-editor/druid/screenshots/images/hello.png)
-![](https://raw.githubusercontent.com/xi-editor/druid/screenshots/images/slider.png)
+[![calc.rs example](https://raw.githubusercontent.com/xi-editor/druid/screenshots/images/calc.png)](./druid/examples/calc.rs)
+[![custom_widget.rs example](https://raw.githubusercontent.com/xi-editor/druid/screenshots/images/custom_widget.png)](./druid/examples/custom_widget.rs)
+[![hello.rs example](https://raw.githubusercontent.com/xi-editor/druid/screenshots/images/hello.png)](./druid/examples/hello.rs)
+[![slider.rs example](https://raw.githubusercontent.com/xi-editor/druid/screenshots/images/slider.png)](./druid/examples/slider.rs)
 
-## Build
+## Concepts
 
-#### Windows
+### druid-shell
 
-run `cargo build`
+The druid toolkit uses druid-shell for a platform-abstracting application shell.
+Druid-shell is responsible for starting a native platform runloop, listening to
+events, converting them into a platform-agnostic representation, and calling a
+user-provided handler with them.
 
-#### macOS
+While druid-shell is being developed with the druid toolkit in mind, it is
+intended to be general enough that it could be reused by other projects
+interested in experimenting with Rust GUI. The druid-shell crate includes a
+couple [non-druid examples].
 
-On macOS, druid requires cairo; see [gtk-rs dependencies] for installation instructions.
+### piet
 
-You may also need to set your `PKG_CONFIG_PATH`; assuming you have installed `cairo` through homebrew, you can build with,
+Druid relies on the [piet 2D graphics abstraction] for drawing. Piet, in turn,
+has multiple backends. Currently: "piet-direct2d", "piet-cairo", and "piet-web".
```

Comment:
I'd probably write this as: "Druid relies on the piet library for drawing and text layout. Piet is a 2D graphics abstraction with multiple backends: `piet-direct2d`, `piet-cairo`, and `piet-web` are currently available, and a GPU backend is planned. In terms of OS support, macOS and Linux use `piet-cairo`, and Windows uses `piet-direct2d`." Feel free to take whatever you like.